packages/protocol/contracts/authorizers/ClientWhitelistRrpAuthorizer.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "./RankedAdminnable.sol";
+import "./interfaces/IClientWhitelistRrpAuthorizer.sol";
+
+/// @title Authorizer contract where client addresses are whitelisted until an
+/// expiration time or indefinitely (until the whitelisting is revoked)
+abstract contract ClientWhitelistRrpAuthorizer is RankedAdminnable, IClientWhitelistRrpAuthorizer {
+  /// @notice Keeps the whitelisting statuses of clients for individual Airnodes
+  mapping(bytes32 => mapping(address => WhitelistStatus)) public airnodeIdToClientAddressToWhitelistStatus;
+
+  /// @notice Called by an admin to extend the whitelist expiration of a
+  /// client
+  /// @param airnodeId Airnode ID from `AirnodeParameterStore.sol`
+  /// @param clientAddress Client address
+  /// @param expirationTimestamp Timestamp at which the client will no longer
+  /// be whitelisted
+  function extendWhitelistExpiration(
+    bytes32 airnodeId,
+    address clientAddress,
+    uint64 expirationTimestamp
```

Comment:
Wouldn't it be more semantic to receive the number of seconds to increase current `expirationTimestamp`? A calling contract could use solidity values like `7 days`, `2 weeks`, etc and we could make the type even smaller (uint32 maybe?).

Reply:
Hmm, good point, and then we wouldn't need the `require()` below. That being said, I expect this to used with a definite deadline, such as "we want to hold a hackathon for this API, so we authorize an admin contract that allows the participant addresses to whitelist any number of their clients, but only until July 30". But then that contract can get the current expiration time and increase it by an amount that clips at July 30. So I suppose the argument should indeed specify the amount to increase the expiration timestamp. Let me think a bit about this and let's also see what @Ashar2shahid thinks.

Reply:
Actually it's fine to keep unix epochs in 32 bits (the only issue being https://en.wikipedia.org/wiki/Year_2038_problem). That being said, using 32/64 bits doesn't matter in this context (wouldn't be cheaper in gas).

Reply:
@bbenligiray wouldn't the clipping have to be done at the dAPP end? 
I think its fine either way, it does make more sense to receive the number of second to extend but then its a different "parameter" from `setWhitelistExpiration`. Could lead to confusion if someone doesn't read what each are.